Fix documentation links

tbekolay opened the pull request:
**Motivation and context:**
@Seanny123 found a broken link in #1421. A bit of grepping revealed a few other broken links. This PR fixes them by using actual Sphinx directives, which we should always use because they would have caught these issues. There is probably also a dead link checker Sphinx extension, but ... meh!

**How has this been tested?**
Built the docs, checked that the links work now.

**How long should this take to review?**

- Quick (less than 40 lines changed or changes are straightforward)

**Types of changes:**

- Non-code change (touches things like tests, documentation, build scripts)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [na] I have included a changelog entry.
- [na] I have added tests to cover my changes.
- [na] I have run the test suite locally and all tests passed.

drasmuss commented:
Sphinx does indeed have a dead link checker, I use it in `nengo-dl` during the doc CI build.  Something like this

`sphinx-build -b linkcheck docs docs/_build -W -D nbsphinx_execute=never`

(`-W` makes it fail on warnings, and `-D nbsphinx_execute=never` means that it doesn't try to build the notebooks during the link checking).  I'm fine not having it in, but just thought I'd save some googling if we do want to add it to nengo as part of this PR :smile:

tbekolay commented:
I added the dead link checking and fixed all the dead or redirected links. One link (https://www.lfd.uci.edu/~gohlke/pythonlibs/) was unfortunately not dead, but could not be reached by `requests`, the library that Sphinx uses to check links. [Another project](https://github.com/SciTools/cartopy/issues/978) compromised by leaving in the link, but not clickable, which seemed an okay compromise, but I can switch to removing the link entirely if that's preferable.
